packages/admin/src/SelectionRoute.tsx

```diff
@@ -39,24 +41,25 @@ export function useSelectionRoute(): [React.ComponentType<IProps>, { id?: string
         [handleSelectId, handleDeselect, handleAdd],
     );
 
-    let selectedId: string | undefined;
-    let selectionMode: "edit" | "add" | undefined;
-    if (match && (match.params.id === "add" || match.params.id?.startsWith("add-"))) {
-        selectedId = match.params.id?.startsWith("add-") ? match.params.id.substr(4) : undefined;
-        selectionMode = "add";
-    } else if (match) {
-        selectedId = match.params.id;
-        selectionMode = "edit";
-    }
+    const routeId: string | null = match && match.params.id ? match.params.id : null;
+    const selection = React.useMemo(() => {
+        let selectedId: string | undefined;
+        let selectionMode: "edit" | "add" | undefined;
+        if (routeId && (routeId === "add" || routeId.startsWith("add-"))) {
+            selectedId = routeId.startsWith("add-") ? routeId.substr(4) : undefined;
+            selectionMode = "add";
+        } else if (routeId) {
+            selectedId = routeId;
+            selectionMode = "edit";
+        }
 
-    return [
-        SelectionRouteInner,
-        {
+        return {
             id: selectedId,
             mode: selectionMode,
-        },
-        api,
-    ];
+        };
+    }, [routeId]);
+
+    return [SelectionRouteInner, selection, api];
```

Comment:
Is it really necessary to have `SelectionRouteInner` in the dependency array?

Reply:
This is not an dependency, this is the React component which the hook returns and is rendered e.g. in EditDialog. Btw, I didn't change the logic here, I just memoized everything the hook returns to avoid unmounting components when the hook returns new values.

Reply:
correct. i have seen this wrong